src/main/java/cubicchunks/CustomCubicChunksWorldType.java

```diff
@@ -44,10 +45,10 @@ public CustomCubicChunksWorldType() {
 		ServerCubeCache cubeCache = world.getCubeCache();
 		
 		// init the worldgen pipeline
-		GeneratorStage terrain = new GeneratorStage("terrain");
-		GeneratorStage surface = new GeneratorStage("surface");
-		GeneratorStage features = new GeneratorStage("features");
-		GeneratorStage population = new GeneratorStage("population");		
+		GeneratorStage terrain = new IndependentGeneratorStage("terrain");
```

Comment:
I don't think these are really independent.

Terrain is independent, but then surface requires cube above it to exist but the cube above can't be in later stage. So this implies that Features stage requires cube below. Then there is lighting which is completely different kind of thing, with it's own quirks and requires 17 blocks from the edge of the cube to exist. And then there is populator which requires cubes from [x, y, z] to [x+1, y+1, z+1].

This is currently done by cancelling generation is dependencies aren't satisfied.